cmd/detectExecuteScan.go

```diff
@@ -98,22 +112,100 @@ func addDetectArgs(args []string, config detectExecuteScanOptions, fileUtils pip
 		args = append(args, fmt.Sprintf("--detect.source.path=%v", config.ScanPaths[0]))
 	}
 
-	mavenArgs, err := maven.DownloadAndGetMavenParameters(config.GlobalSettingsFile, config.ProjectSettingsFile, fileUtils, httpClient)
-	if err != nil {
-		return nil, err
-	}
+	// if buildCode is false then user may provide a path to a local .m2 config where previously build artifacts are present
+	// when buildCode is true (default behaviour) we build the code
+	if !config.BuildCode {
+		mavenArgs, err := maven.DownloadAndGetMavenParameters(config.GlobalSettingsFile, config.ProjectSettingsFile, fileUtils, httpClient)
 
-	if len(config.M2Path) > 0 {
-		absolutePath, err := fileUtils.Abs(config.M2Path)
 		if err != nil {
 			return nil, err
 		}
-		mavenArgs = append(mavenArgs, fmt.Sprintf("-Dmaven.repo.local=%v", absolutePath))
+
+		if len(config.M2Path) > 0 {
+			absolutePath, err := fileUtils.Abs(config.M2Path)
+			if err != nil {
+				return nil, err
+			}
+			mavenArgs = append(mavenArgs, fmt.Sprintf("-Dmaven.repo.local=%v", absolutePath))
+		}
+
+		if len(mavenArgs) > 0 {
+			args = append(args, fmt.Sprintf("\"--detect.maven.build.command='%v'\"", strings.Join(mavenArgs, " ")))
+		}
+	} else {
+		c1 := command.Command{}
+		switch config.BuildTool {
+		case "maven", "mta":
+			mavenBuildCommand := []string{"clean", "install", "-DskipTests=true"}
+			mavenBuildArgs, err := maven.DownloadAndGetMavenParameters(config.GlobalSettingsFile, config.ProjectSettingsFile, fileUtils, httpClient)
+			if err != nil {
+				return nil, err
+			}
+			if len(config.M2Path) > 0 {
+				absolutePath, err := fileUtils.Abs(config.M2Path)
+				if err != nil {
+					return nil, err
+				}
+				mavenBuildArgs = append(mavenBuildArgs, fmt.Sprintf("-Dmaven.repo.local=%v", absolutePath))
+			}
+			mavenBuildCommand = append(mavenBuildCommand, mavenBuildArgs...)
+			pomFiles, err := newUtils().Glob(filepath.Join("**", "pom.xml"))
+			if err != nil {
+				log.Entry().Info("Build tool is " + config.BuildTool + " and no pom xml found. Detect scan will proceed without a build")
+			}
+			//When pom.xml is present in the workspace directory (parent pom) and if this has not been added to BuildDescriptorExcludeList we build the code
+			//if not then we find every pom xml in the current workspace and use it to build unless it is added in BuildDescriptorExcludeList*/
+			if findElement(pomFiles, "pom.xml") && !findElement(config.BuildDescriptorExcludeList, "pom.xml") {
+				args = append(args, fmt.Sprintf("\"--detect.maven.build.command='%v'\"", strings.Join(mavenBuildCommand, " ")))
+			} else {
+				localMavenBuild(fileUtils, config, &c1, pomFiles)
+			}
+		default:
+			log.Entry().Info("Detect scan will proceed without a build")
```

Comment:
The log-message could be misleading. You should mention that this is because the build-tool is not supported, not because of the new `buildCode` parameter. Something like:

```suggestion
			log.Entry().Infof("Unsupported buildTool: '%s' - Detect scan will proceed without a build, even though buildCode = true", config.BuildTool)
```